src/features/soulkeys/SoulKeyWizard.tsx

```diff
@@ -0,0 +1,133 @@
+import { useWalletStatus } from 'features/auth';
+import { chain } from 'features/cosoul/chains';
+import { useNavQuery } from 'features/nav/getNavData';
+import { client } from 'lib/gql/client';
+import { useQuery } from 'react-query';
+
+import { Check, Square } from '../../icons/__generated';
+import { useWeb3React } from 'hooks/useWeb3React';
+import { Flex, Panel, Text } from 'ui';
+
+const Step = ({ label, test }: { label: string; test?: boolean }) => {
+  return (
+    <Flex css={{ justifyContent: 'space-between' }}>
+      <Text>{label}</Text>
+      {test ? <Check color="complete" /> : <Square color="warning" />}
+    </Flex>
+  );
+};
+export const SoulKeyWizard = () => {
+  const { data } = useNavQuery();
+  const name = data?.profile.name;
+  const { chainId } = useWeb3React();
+  const { address } = useWalletStatus();
+  const onCorrectChain = chainId === Number(chain.chainId);
+  const hasName = name && !name.startsWith('New User') && !!address;
+  const hasCoSoul = !!data?.profile.cosoul;
+
+  const { data: subjectProfile } = useQuery(
+    ['soulKeys', address, 'profile'],
```

Comment:
@wingfeatherwave check out Caching Examples and docs for queryKey 
https://tanstack.com/query/v5/docs/react/guides/query-keys
https://tanstack.com/query/v5/docs/react/guides/caching

Reply:
lets change this key to be more structured and useful for invalidation